packages/graphback-datasync/src/providers/DataSyncConflictProvider.ts

```diff
@@ -0,0 +1,122 @@
+import { ModelDefinition, GraphbackContext, TransformType, FieldTransform, NoDataError, GraphbackOperationType } from '@graphback/core';
+import { DataSyncModelConflictConfig, DataSyncFieldNames, ConflictMetadata, ConflictError } from '../util';
+import { MongoDeltaSource } from '../deltaSource';
+import { DataSyncMongoDBDataProvider } from './DatasyncMongoDBDataProvider';
+
+export const MAX_RETRIES = 3;
+
+/**
+ * Data Provider with update conflicts and optional conflict resolution
+ * that connects to the MongoDB database
+ */
+export class DataSyncConflictMongoDBDataProvider<Type = any> extends DataSyncMongoDBDataProvider<Type> {
+  protected conflictConfig: DataSyncModelConflictConfig
+  protected deltaSource: MongoDeltaSource
+
+  public constructor(model: ModelDefinition, client: any, dataSyncConflictConfig: DataSyncModelConflictConfig) {
+    super(model, client);
+    this.conflictConfig = dataSyncConflictConfig;
+
+    if (!this.conflictConfig.conflictResolution?.resolveUpdate) {
+      this.conflictConfig.conflictResolution = {
+        resolveUpdate(conflict: ConflictMetadata) {
+          throw new ConflictError(conflict)
```

Comment:
Let's disscuss. We talking here about our users holding GB'S of data in diff table with still client side resolution by default. 

IMHO server side should be default, but would love to get your opinion as well

Reply:
> IMHO server side should be default, but would love to get your opinion as well

Onboard with this